### DISCUSSION TITLE
How do I use external ELK with "sensor" T-Pot installation type?

### DISCUSSION THREAD

**Discussion by anna-oake** [telekom-security/tpotce#897] [Q&A]
I've used T-Pot in a "standard" way for some time, but now I have my own ELK stack set up and would like to switch T-Pot to this cluster.

To do this I've reinstalled T-Pot using the "sensor" installation type to get rid of internal ELK.

How do I make T-Pot push logs to my external elasticsearch server?

**Comment by Ganesh-403** [comment]
@anna-oake 
The **`SENSOR`** installation type is **not intended to send logs to an arbitrary external Elasticsearch cluster directly**.

From the official documentation and configuration, a Sensor is designed to forward its logs to a **T-Pot Hive** deployment, which provides the central Logstash/Elasticsearch/Kibana stack. The `.env` configuration for a Sensor includes settings such as:

- `TPOT_TYPE=SENSOR`
- `TPOT_HIVE_USER`
- `hive.crt`
- SSL verification settings

and the deployment instructions explicitly describe connecting the Sensor to a Hive instance. :contentReference[oaicite:0]{index=0}

So if your goal is:

> **Sensor → External Elasticsearch**

there isn't a documented configuration option to simply specify:

```text
ELASTICSEARCH_HOST=<your-cluster>
```

or similar.

### What are your options?

**1. Recommended: Use a T-Pot Hive**

This is the deployment model officially supported by T-Pot:

```
Sensor(s)
     │
     ▼
Logstash (Hive)
     ▼
Elasticsearch
     ▼
Kibana
```

The Sensor securely ships its logs to the Hive, which then indexes them. :contentReference[oaicite:1]{index=1}

---

**2. Integrate with your own ELK stack (custom solution)**

If you already have your own Elasticsearch cluster, you'll likely need to customize the Logstash pipeline on the Sensor so that its output points to your Elasticsearch instance instead of a Hive.

Conceptually, the Logstash output would look something like:

```ruby
output {
  elasticsearch {
    hosts => ["https://your-es:9200"]
    user => "..."
    password => "..."
    ssl => true
  }
}
```

However, **from what I can tell, this is not a documented or officially supported T-Pot deployment model**. You would also be responsible for:

- installing the appropriate index templates/ILM policies,
- creating mappings,
- importing dashboards,
- and ensuring compatibility with T-Pot's ingest pipelines. T-Pot normally manages these components as part of its own stack. :contentReference[oaicite:2]{index=2}

### A couple of questions

To suggest the best approach, it would help to know:

1. Is your external ELK stack **self-managed Elasticsearch + Kibana**, or **Elastic Cloud**?
2. Do you want to keep **all** of T-Pot's dashboards and visualizations, or do you only need the raw honeypot events indexed into your existing cluster?

If you simply want to centralize multiple T-Pot deployments while retaining the built-in dashboards and management, I'd recommend using the supported **Hive + Sensor** architecture. If your goal is to integrate with an existing enterprise ELK deployment, it should be possible via Logstash customization, but it will require manual configuration and isn't something the T-Pot documentation currently describes as a supported workflow.

If this solves your problem, feel free to mark it as the accepted answer so others can find it easily.
